index.html

```diff
@@ -1,11 +1,215 @@
+      }
+      return str;
+    }
+
+    function formatHtmlItem(label, item, noNewLine) {
+      var desc = "";
+      if ("description" in item) {
+        desc = " - " + item["description"];
+      }
+      var exp = "";
+      if ("explain" in item) {
+        exp = ' <a href="#"><span class="glyphicon glyphicon-comment" title="' + item["explain"] + '"></span></a>';
+      }
+      var str = `<li><a href='${item["url"]}'>${label}</a>${desc} ${exp}</li>`;
+
+      if (undefined === noNewLine) {
+        return str + "";
+      }
+      return str;
+    }
+  </script>
+</head>
+
+<body>
+<div id="app">
+  <div class="section title">
+    <h1>Saritasa recommended packages for PHP</h1>
+  </div>
+
+  <div class="section tag-cloud">
+    <div>Tag Cloud</div>
+  </div>
+
+  <div class="section search-form">
+    <input type="text" id="keyword" placeholder="Search keyword" v-model="keyword">
+    <span class="label label-success selected-tag hidden">Tag 1<a><i class="remove glyphicon glyphicon-remove-sign glyphicon-white"></i></a></span>
+    <p>{{ keyword }}</p>
+  </div>
+
+  <div class="section search-result">
+
+  </div>
+
+  <script>
+    new Vue({
+      el: '#app',
+      data: {
+        keyword: '',
+      },
+      methods: {
+        formatMarkdown: function () {
+          var markdownString = "";
+          var yamlData = yamlFile;
+          Object.keys(yamlData).forEach(function(key1){
+            // Level 1
+            markdownString += level1(key1);
+            // Level 2
+            Object.keys(yamlData[key1]).forEach(function(key2){
+              markdownString += level2(key2);
+
+              // Level 3
+              var languageGroups = new Object();
+              Object.keys(yamlData[key1][key2]).forEach(function(key3){
+                var language = "Other";
+                if ("language" in yamlData[key1][key2][key3]) {
+                  language = yamlData[key1][key2][key3]["language"];
+                }
+                if (!(language in languageGroups)) {
+                  languageGroups[language] = {};
+                }
+                // Check url of websites section
+                if (key2 === "websites" && !("url" in yamlData[key1][key2][key3])) {
+                  yamlData[key1][key2][key3]["url"] = "http://" + key3;
+                }
+
+                languageGroups[language][key3] = yamlData[key1][key2][key3];
+              });
+
+              // Show items
+              Object.keys(languageGroups).forEach(function(keyLang){
+                if (key2 !== "packages") {
+                  markdownString += level3("_" + keyLang + "_");
+                }
+                Object.keys(languageGroups[keyLang]).forEach(function(keyItem) {
+                  markdownString += formatItem(keyItem, languageGroups[keyLang][keyItem]);
+                });
+              });
+            });
+          });
+
+          return markdownString;
+        },
+        formatHtml: function () {
```

Comment:
One of primary requirements was to make a VueJS app - which:

* uses packages installation via NPM (rather than just reference 3d party files on CDN)
* split code into components (ex. WebSite.vue, Article.vue, Package.vue, WebSitesList.vue, ArticlesList.vue, PackagesList.vue, ResourceSections.vue) each of them can accept data of specific type and render it - instead of one huge file for everything, which contains JS code, interleaving with HTML markup
* use VueJS template engine instead of manual building HTML by strings concatenation - intention is to separate JS from HTML markup (by using VueJS as true MVC framework).
* avoid jQuery - it is almost not needed with VueJS

Just like we do VueJS single-page applications in production projects.

Finally we must have:
* package.json that defines project dependencies
* set of VueJS components, each of them handles its specific part of page ex.:
  * WebSite.vue - knows, how to draw single link to website, how to display its description, how to build URL from website title, etc.
  * Article.vue - knows, how to draw link to article, validates data, shows warning, if not all expected data present - ex. no URL provided
  * Package.vue - knows, how to draw link to Package, validates data
  * Above atomic components can have one parent or extend same template
  * WebSitesList.vue - ex. handles event, that list of websites to display is updated, and renders list of websites, using WebSite.vue component. If no sites to display, hides itself (including title of "Websites" section). Knows, that sites may be in different languages and displays sites in corresponding sections.
  * ArticlesList.vue - does same with articles list
  * PackagesList.vue - does same with packages list, but doesn't take care about languages, displays single list of packages
  * ResourceSections.vue - just displays PackagesList, ArticlesList, WebSites list in correct order
  * TagsCloud.Vue - draws list of tags, matching currently filtered elements. emits event, that user selected tag, when user clicks on it - "SelectedTagsListChanged"
  * SelectedTags - draws list of currently selected events, allows to remove them, emits corresponding event - "SelectedTagsListChanged"
  * SearchBox.vue - contains search box input, handles user input, emits corresponding event (probably, debounces input)
  * FilterControls.vue - combines TagsCloud, SelectedTags, SearchBox components
  * Page.vue - combines FilterControls and ResourceSections, routes events, like "SelectedTagsListChanged", "SearchInputChanged", updates data in Vuex - which resources must be displayed right now, after user input changed.
* WebPack settings, that allow to build Vue components to JS (you will get it "for free" when you create vue app, as any VueJS tutorial says).
* index.html file, that only initializes JS application.

Final structure of components may be different (I didn't list here technology stack chooser - select between JS and PHP, etc.).